hcl2template/parser.go

```diff
@@ -210,12 +210,45 @@ func (p *Parser) Parse(filename string, varFiles []string, argVars map[string]st
 	{
 		hclVarFiles, jsonVarFiles, moreDiags := GetHCL2Files(filename, hcl2AutoVarFileExt, hcl2AutoVarJsonFileExt)
 		diags = append(diags, moreDiags...)
+
+		var variableFiles []*hcl.File
+
+		for _, jsonFile := range jsonVarFiles {
+			f, moreDiags := p.ParseJSONFile(jsonFile)
+			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
+			variableFiles = append(variableFiles, f)
+		}
+
+		for _, hclFile := range hclVarFiles {
+			f, moreDiags := p.ParseHCLFile(hclFile)
+			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
+			variableFiles = append(variableFiles, f)
+		}
+
+		// We are trying to maintain the user given order for the var files with this for loop.
 		for _, file := range varFiles {
 			switch filepath.Ext(file) {
 			case ".hcl":
-				hclVarFiles = append(hclVarFiles, file)
+				f, moreDiags := p.ParseHCLFile(file)
+				diags = append(diags, moreDiags...)
+				if moreDiags.HasErrors() {
+					continue
+				}
+				variableFiles = append(variableFiles, f)
 			case ".json":
-				jsonVarFiles = append(jsonVarFiles, file)
+				f, moreDiags := p.ParseJSONFile(file)
+				diags = append(diags, moreDiags...)
+				if moreDiags.HasErrors() {
```

Comment:
For the next reroll, do we really need to do this? If there was an error during this pass, we are going to stop processing the configuration eventually, what are we trying to avoid by continuing to the next iteration?
If `f` is `nil` as a result of the processing failure, I would think we can encapsulate this `append` statement around a `if f != nil` instead of trying to continue to the next iteration here.